Add towncrier for automated release notes (#2384)
nautobot/nautobot: nautobot/docs/development/release-checklist.md

### Update the Changelog

Create a release branch off of `develop` (`git checkout -b release-1.4.3 develop`)

> Generate release notes with `towncrier build --version 1.4.3`. This will update the release notes in `nautobot/docs/release-notes/version-1.4.md`, stage that file in git and `git rm` all of the fragments if you answer yes to the towncrier prompt `Is it okay if I remove those files? [Y/n]:`.

Author: It looks like the `poetry version <type>` only updates the pyproject.toml so you would have to run `poetry install` for towncrier to pick up the new version.

### Update the Changelog

Create a release branch off of `develop` (`git checkout -b release-1.4.3 develop`)

> Generate release notes with `towncrier build --version 1.4.3`. This will update the release notes in `nautobot/docs/release-notes/version-1.4.md`, stage that file in git and `git rm` all of the fragments if you answer yes to the towncrier prompt `Is it okay if I remove those files? [Y/n]:`.

Reviewer: ```suggestion
Generate release notes with `towncrier build --version 1.4.3` and answer `yes` to the prompt `Is it okay if I remove those files? [Y/n]:`. This will update the release notes in `nautobot/docs/release-notes/version-1.4.md`, stage that file in git, and `git rm` all of the fragments that have now been incorporated into the release notes.
```

Run `invoke markdownlint` to make sure the generated release notes pass the linter checks.

> Check the git diff to verify the changes are correct (`git diff --cached`)

Reviewer: ```suggestion
Check the git diff to verify the changes are correct (`git diff --cached`).
```
